sonar-core/src/main/resources/org/sonar/core/qualityprofile/db/QualityProfileMapper.xml

```diff
@@ -151,6 +151,14 @@
     </where>
   </select>
 
+  <select id="countProjectsByProfile" resultType="org.sonar.core.qualityprofile.db.QualityProfileProjectCount">
```

Comment:
Indeed explicit INNER is preferred as I don't believe the default behaviors of all our supported databases.